src/service/git/gitlab/gitlab-client.ts

```diff
@@ -45,10 +45,24 @@ export default class GitLabClient implements GitClient {
   // READ
 
   // example: <host>/api/v4/projects/<namespace>%2Fbackporting-example/merge_requests/1
-  async getPullRequest(namespace: string, repo: string, mrNumber: number, squash = true): Promise<GitPullRequest> {
+  async getPullRequest(namespace: string, repo: string, mrNumber: number, squash: boolean | undefined): Promise<GitPullRequest> {
     const projectId = this.getProjectId(namespace, repo);
     const { data } = await this.client.get(`/projects/${projectId}/merge_requests/${mrNumber}`);
 
+    if (squash === undefined) {
+      if (data.state == "opened") {
+        this.logger.debug("cherry-pick all commits because the pull request is open");
+        squash = false;
+      } else {
+        squash = data.squash_commit_sha ? true : false;
```

Comment:
Did not check if that field can have other values like `null` but if it can be set or `undefined`, this could simplify the instruction. Just a minor.
```suggestion
        squash = data.squash_commit_sha !== undefined;
```